ttnn/cpp/ttnn/global_circular_buffer.cpp

```diff
@@ -0,0 +1,57 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "global_circular_buffer.hpp"
+
+#include <memory>
+#include "tt_metal/impl/buffers/global_circular_buffer.hpp"
+#include "tt_metal/include/tt_metal/global_circular_buffer.hpp"
+
+namespace ttnn::global_circular_buffer {
+
+MultiDeviceGlobalCircularBuffer::MultiDeviceGlobalCircularBuffer(MeshDevice* mesh_device) {
+    TT_ASSERT(
+        mesh_device != nullptr,
+        "Must provide a valid mesh_device when initializing a global circular buffer on multiple devices.");
+    auto devices = mesh_device->get_devices();
```

Comment:
It won't be a copy since the get_devices function returns a new vector each invocation, though we shouldn't really assume/depend on that. I'll update it.